### PR TITLE
01 project add classes

### PR DESCRIPTION
**Pull request by gak92**
In this pull request:

- [X] Add the `Person` class and its constructor and methods
- [X] Add the `Student` class which inherit from the Person class and add its constructor and methods
- [X] Add the `Teacher` class which inherit from the Person class and add its constructor and methods

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def initialize(id, age, name = 'Unknown', parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || parent_permission
```
Comment (DammyShittu) [inline person.rb]: - [ ] I think you should use the variable `@parent_permission`, instead of `parent_permission`.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @gak92,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] For more professional readme documentation, kindly populate the different sections on your readme with meaningful information

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (gak92) [reply person.rb]: @DammyShittu,
Thanks for the review.
I just correct the mistake.

**Review by gemmen29 — APPROVED** [review]
Hi @gak92, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gak92** [comment]
@gemmen29,
Thanks for the review

**Pull request merged**